lib/data/lcu/lcu.dart

```diff
@@ -149,6 +156,21 @@ class LCU {
       }
     }
 
+    if (directory.path.endsWith(LCU.macosFileType)) {
+      try {
+        final contentFolder = await directory.list().firstWhere((item) => item.path.endsWith('Contents'));
+        final lolFolder = await (contentFolder as Directory).list().firstWhere((item) => item.path.endsWith('LoL'));
+
+        await for (final FileSystemEntity f in (lolFolder as Directory).list()) {
+          checkLockfile(f);
+        }
+      } catch (_) {}
+    } else {
+      await for (final FileSystemEntity f in directory.list()) {
+        checkLockfile(f);
+      }
+    }
+
```

Comment:
Can you please rollback and separate a macos implementation from windows into `_getLockfileFromMacosLolDirectory` then add Platform check in `saveLockfileDirectory`